backend/src/data/repositories/chat-message/chat-message.repository.ts

```diff
@@ -0,0 +1,109 @@
+import { SortOrder } from '~/common/enums/enums';
+import {
+  ChatMessageCreateRequestDto,
+  ChatMessageGetAllItemResponseDto,
+  ChatMessageGetRequestDto,
+} from '~/common/types/types';
+import {
+  ChatMessage as ChatMessageM,
+  MenteesToMentors as MenteesToMentorsM,
+} from '~/data/models/models';
+
+type Constructor = {
+  ChatMessageModel: typeof ChatMessageM;
+  MenteesToMentors: typeof MenteesToMentorsM;
+};
+
+class ChatMessage {
+  #ChatMessageModel: typeof ChatMessageM;
+
+  #MenteesToMentors: typeof MenteesToMentorsM;
+
+  public constructor({ ChatMessageModel, MenteesToMentors }: Constructor) {
+    this.#ChatMessageModel = ChatMessageModel;
+    this.#MenteesToMentors = MenteesToMentors;
+  }
+
+  public getAll({
+    userId,
+    chatOpponentId,
+  }: ChatMessageGetRequestDto): Promise<ChatMessageGetAllItemResponseDto[]> {
+    return this.#ChatMessageModel
+      .query()
+      .select()
+      .where({ senderId: userId, receiverId: chatOpponentId })
+      .orWhere({ senderId: chatOpponentId, receiverId: userId })
+      .withGraphJoined(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto[]>()
+      .execute();
+  }
+
+  public create({
+    senderId,
+    receiverId,
+    message,
+  }: ChatMessageCreateRequestDto): Promise<ChatMessageGetAllItemResponseDto> {
+    return this.#ChatMessageModel
+      .query()
+      .insert({ senderId, receiverId, message })
+      .withGraphFetched(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto>()
+      .execute();
+  }
+
+  public async getAllLastMessages(
+    userId: number,
+  ): Promise<ChatMessageGetAllItemResponseDto[]> {
+    const usersMentors = await this.#MenteesToMentors
```

Comment:
But what if I have to make queries which do not exist in MenteesToMentors repo? If I create them in that repo, thay seem to be redundant there because they are used only in one place

Reply:
Even though they are used only in one place, MenteesToMentors repo is the correct place to write queries to that model
We have lots of `getAll` methods, which are used only in one place (in service) :)